### DISCUSSION TITLE
dbus will not start, crashes with launchctl error

### DISCUSSION THREAD

**Discussion by scottwn** [Homebrew/discussions#2318] [Everyday usage] [answered]
I'm trying to get [scli](https://github.com/isamert/scli) to work on a new M1 MacBook. I've used scli successfully on both Mac and Linux machines before.

scli depends on dbus, but I can not get dbus to start. This error is persistent.
```
~ $ brew services start dbus
Bootstrap failed: 5: Input/output error
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/scott/Library/LaunchAgents/org.freedesktop.dbus-session.plist` exited with 5.
```

I've confirmed that the environment variables are set correctly as suggested on this page https://github.com/AsamK/signal-cli/wiki/DBus-service#user-content-dbus-on-macos

I've reloaded the plist successfully as suggested here https://developer.apple.com/forums/thread/665661

This stackexchange question seems to be the same issue that I'm having, but no answers yet. https://apple.stackexchange.com/questions/417707/dbus-daemon-can-not-started?newreg=d985bef32e9f48adaf3b6b68ed82e9f4

Can anyone help get dbus running on an M1 Mac?

**Comment by SMillerDev** [comment, marked as answer]
The plist name and label should be determined by homebrew, otherwise this won't work.

**Reply by scottwn** [reply]
Is that not done by `brew install dbus`? How do I make sure that Homebrew is setting the plist name?

**Reply by scottwn** [reply]
Not sure why, but stopping and starting the service worked this time. I had tried this several times and the error persisted, but this time it worked.... strange
```
~ [1]$ brew services start dbus
Bootstrap failed: 5: Input/output error
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/scott/Library/LaunchAgents/org.freedesktop.dbus-session.plist` exited with 5.
~ [1]$ brew services stop dbus
Stopping `dbus`... (might take a while)
==> Successfully stopped `dbus` (label: org.freedesktop.dbus-session)
~ $ brew services start dbus
==> Successfully started `dbus` (label: org.freedesktop.dbus-session)
```

**Reply by vectorsss** [reply]
Hi, Have you fixed this problem? I got the same problem with Intel Mac.

**Comment by johnhamelink** [comment]
Hi, I had a similar issue, but for me the error I was getting back was:

```
❯ brew services run dbus
Bootstrap failed: 5: Input/output error
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /opt/homebrew/opt/dbus/homebrew.mxcl.dbus.plist` exited with 5.
❯ /bin/launchctl bootstrap gui/501 /opt/homebrew/opt/dbus/homebrew.mxcl.dbus.plist
```

Now when I visited `/opt/homebrew/opt/dbus/`, I found that the plist file was called `/opt/homebrew/opt/dbus/org.freedesktop.dbus-session.plist`. I manually ran `/bin/launchctl bootstrap gui/501 /opt/homebrew/opt/dbus/org.freedesktop.dbus-session.plist`, and then I could run `launchctl start org.freedesktop.dbus-session` successfully! I planned on contributing this back as a fix, as I was left with the impression the Formulae was out of date, however when I checked, the plist name is correct. I'm not sure how I was left with this artifact, but in case this happens to someone else I thought I'd leave it here.

**Reply by s1gtrap** [reply]
I'm experiencing the exact same issue except your workaround doesn't seem to do anything for me.

What is `launchctl bootstrap gui/501 /Users/s1g/Library/LaunchAgents/org.freedesktop.dbus-session.plis` supposed to do? `ps ax | grep dbus` doesn't appear any different after.
